scripts/eurostat/birth_death_migration/import_data.py

```diff
@@ -0,0 +1,131 @@
+    # Replace value column with extra space.
+    preprocessed_df.value = preprocessed_df.value.str.replace(
+        "([0-9:])$", lambda m: m.group(0) + ' ')
+    statistical_variable = None
+    # In case this returns only one element in the list.
+    first_column_list = preprocessed_df.columns[0].rsplit(sep=",", maxsplit=1)
+    if len(first_column_list) == 2:
+        statistical_variable, geo = first_column_list
+    else:
+        geo = first_column_list[0]
+    geo = geo.replace(r'\time', '')
+    assert geo == "geo", "Expected the column header to end with 'geo'."
+
+    if statistical_variable:
+        split_df = preprocessed_df[preprocessed_df.columns[0]].str.rsplit(
+            ",", n=1, expand=True)
+        preprocessed_df['statistical_variable'] = split_df[0]
+        preprocessed_df['geo'] = split_df[1]
+        preprocessed_df.drop(columns=[preprocessed_df.columns[0]], inplace=True)
+
+        preprocessed_df = (preprocessed_df.set_index(["geo", "time"]).pivot(
+            columns="statistical_variable")['value'].reset_index().rename_axis(
+                None, axis=1))
+    # Fill missing 'geo' values with a colon.
+    preprocessed_df.fillna(': ', inplace=True)
+
+    # Split notes out of values.
+    # Ex: "5598920 b" -> "5598920", "b".
+    # Ex: "5665118" -> "5665118", ""
+    # Ex: ": c" -> "", "c"
+    # Append extra space for all values that do not come with a note.
+    preprocessed_df[preprocessed_df.columns[2:]] = preprocessed_df[
+        preprocessed_df.columns[2:]].replace("([0-9:])$",
+                                             lambda m: m.group(0) + ' ')
+    # replace comma in column names with a vertical bar so that we can save it in csv later
+    new_column_names = []
+    for column_name in preprocessed_df.columns:
+        new_column_names.append(column_name.replace(',', '|'))
+    preprocessed_df.columns = new_column_names
+    for column_name in preprocessed_df.columns[2:]:
+        preprocessed_df['{0}'.format(column_name)], preprocessed_df[
+            '{0}|notes'.format(column_name)] = (zip(
+                *preprocessed_df[column_name].apply(lambda x: x.split(' '))))
+    return preprocessed_df
+
+
+def clean_data(preprocessed_df, output_path):
+    """Drops unnecessary columns that are not needed for data import and reformat column names."""
+    if preprocessed_df is None:
+        raise ValueError("Uninitialized value of processed data frame. "
+                         "Please check you are calling preprocess_data "
+                         "before clean_data.")
+    # drop unused columns
+    clean_df = preprocessed_df[preprocessed_df.columns[:len(
+        preprocessed_df.columns
+    ) // 2 + 1]]  # number of columns should be 2 + 2X, we want the first 2 + X
+    # columns
+    # replace colon with NaN.
+    clean_df = clean_df.replace(':', '')
+    clean_df.to_csv(output_path, index=False)
+    clean_df = pd.read_csv(output_path)
```

Comment:
What is happening here? you do another to_csv later too.

Reply:
Thanks for the catch. I must have accidentally committed this when doing the testing